lib/ui/home/home.dart

```diff
@@ -67,66 +66,68 @@ class _HomeState extends State<Home> {
-          break;
-        case 'dining':
-          orderedCards.add(DiningCard());
-          break;
-        case 'news':
-          orderedCards.add(NewsCard());
-          break;
-        case 'events':
-          orderedCards.add(EventsCard());
-          break;
-        case 'weather':
-          orderedCards.add(WeatherCard());
-          break;
-        case 'availability':
-          orderedCards.add(AvailabilityCard());
-          break;
-        case 'schedule':
-          orderedCards.add(ClassScheduleCard());
-          break;
-        case 'finals':
-          orderedCards.add(FinalsCard());
-          break;
-        case 'MyUCSDChart':
-          orderedCards.add(MyUCSDChartCard());
-          break;
-        case 'student_id':
-          orderedCards.add(StudentIdCard());
-          break;
-        case 'staff_info':
-          orderedCards.add(StaffInfoCard());
-          break;
-        case 'student_info':
-          orderedCards.add(StudentInfoCard());
-          break;
-        case 'parking':
-          orderedCards.add(ParkingCard());
-          break;
-        case 'staff_id':
-          orderedCards.add(StaffIdCard());
-          break;
-        case 'shuttle':
-          orderedCards.add(ShuttleCard());
-          break;
+      if (!webCards.containsKey(card)) {
+        switch (card) {
+          case 'QRScanner':
+            orderedCards.insert(0, ScannerCard());
+            break;
+          case 'NativeScanner':
+            orderedCards.insert(0, NativeScannerCard());
+            break;
+          case 'MyStudentChart':
+            orderedCards.add(MyStudentChartCard());
+            break;
+          case 'student_survey':
+            orderedCards.add(SurveyCard());
+            break;
+          case 'dining':
+            orderedCards.add(DiningCard());
+            break;
+          case 'news':
+            orderedCards.add(NewsCard());
+            break;
+          case 'events':
+            orderedCards.add(EventsCard());
+            break;
+          case 'weather':
+            orderedCards.add(WeatherCard());
+            break;
+          case 'availability':
+            orderedCards.add(AvailabilityCard());
+            break;
+          case 'schedule':
+            orderedCards.add(ClassScheduleCard());
+            break;
+          case 'finals':
+            orderedCards.add(FinalsCard());
+            break;
+          case 'MyUCSDChart':
+            orderedCards.add(MyUCSDChartCard());
+            break;
+          case 'student_id':
+            orderedCards.add(StudentIdCard());
+            break;
+          case 'parking':
+            orderedCards.add(ParkingCard());
+            break;
+          case 'shuttle':
+            orderedCards.add(ShuttleCard());
+            break;
+        }
+      }
+      else {  // dynamically insert webCards into the list
+        orderedCards.add(
+            WebViewContainer(
+                titleText: CardTitleConstants.titleMap[card],
```

Comment:
You are referencing a constant that is housed within the mobile app source code. This means that an update via the app store and google play store is required anytime we add a new card.